ci_cd/utils/versions.py

```diff
@@ -758,8 +828,31 @@ def update_specifier_set(  # pylint: disable=too-many-statements
             # does not need updating. To communicate this, make updated_specifiers
             # non-empty, but include only an empty string.
             updated_specifiers.append("")
+
+    elif (
+        latest_version.python_version
+        and latest_version.as_python_version().epoch not in current_version_epochs
+    ):
+        # The latest version is *not* included in the specifier set.
+        # And the latest version is NOT in the same epoch as the current version range.
+
+        # Sanity check that the latest version's epoch is larger than the largest
+        # epoch in the specifier set.
+        if current_version_epochs and latest_version.as_python_version().epoch < max(
+            current_version_epochs
+        ):
+            raise UnableToResolve(
+                "The latest version's epoch is smaller than the largest epoch in "
```

Comment:
should be covered by tests

Reply:
Is it fine with you to reference #211 and say that's that for now?